### PR TITLE
fix: update apq middleware build

### LINKED ISSUE #819: Is APQ working?
### Package

`graphql-hooks`

### Description

We recently release support for APQ and during the publishing of the package several warnings were generated:

```
(!) Unresolved dependencies
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
@aws-crypto/sha256-universal (imported by src\middlewares\apqMiddleware.js)
(!) Mixing named and default exports
https://rollupjs.org/guide/en/#outputexports
The following entry modules are using named and default exports together:
src\middlewares\apqMiddleware.js

Consumers of your bundle will have to use chunk['default'] to access their default export, which may not be what you want. Use `output.exports: 'named'` to disable this warning
created lib/middlewares/apqMiddleware.js in 260ms
```

We should try whether using this middleware from an app actually works.

### PR DESCRIPTION
**Pull request by ianlovell-nearform**
- Replaces `@aws-crypto/sha256-universal` with `@aws-crypto/sha256-browser`
- `APQMiddleware` now a named export
- Replaces TestCafe with Cypress
- Runs acceptance tests on CI

resolves #819

### CODE REVIEW

**Comment by simoneb** [comment]
@rp4rk is looking into https://github.com/nearform/graphql-hooks/issues/838 which has also to do with typings, let's make sure we come to an organic solution

**Comment by rp4rk** [comment]
This LGTM so far

File: examples/create-react-app/src/App.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import React, { useState } from 'react'
 import { GraphQLClient, ClientContext } from 'graphql-hooks'
+// eslint-disable-next-line no-unused-vars
```
Comment (rp4rk) [inline examples/create-react-app/src/App.js:3]: Was including this intended?

File: packages/graphql-hooks/rollup.config.js [hunk 2]
```diff
@@ -14,5 +15,12 @@ export default generateRollupConfig({
     format: 'cjs',
     indent: false
   },
-  plugins: [esbuild(), babel(), sizeSnapshot()]
+  plugins: [
+    esbuild(),
+    replace({
```
Comment (rp4rk) [inline packages/graphql-hooks/rollup.config.js:20]: I can't find reference to `process.env.NODE_ENV` outside of some configuration files, is this vestigial?

**Review by rp4rk — APPROVED** [review]
Couple of things I've noticed but I don't think any of them are huge blockers at all if they prove to be

File: examples/create-react-app/src/App.js (hunk 1 above)
Comment (ianlovell-nearform) [reply examples/create-react-app/src/App.js:3]: yeah, the APQ issue with `crypto` not being polyfilled with webpack 5 only surfaces when the middleware is imported, once #847 is complete, that example could be used for the acceptance tests

**Pull request merged**